tests/conftest.py

```diff
@@ -29,41 +29,51 @@ def node(host, request):
         'mimic': 13,
         'dev': 99
     }
-    if not request.node.get_marker(node_type) and not request.node.get_marker('all'):  # noqa E501
-        pytest.skip("Not a valid test for node type: %s" % node_type)
 
-    if request.node.get_marker("no_lvm_scenario") and lvm_scenario:
+    # capture the initial/default state
+    test_is_applicable = False
+    for marker in request.node.iter_markers():
+        if marker.name not in group_names and marker.name != 'all':
+            test_is_applicable = False
```

Comment:
you can't have this line here, because you could be reverting a `test_is_applicable` being `True`. Since this is a loop you might get True, True, False